consumer.go

```diff
@@ -798,7 +798,13 @@ func (c *Consumer) isPotentialExtraTopic(topic string) bool {
 }
 
 func (c *Consumer) refreshMetadata() error {
-	err := c.client.RefreshMetadata()
+	var topics []string
+
+	if !c.client.config.Metadata.Full {
+		topics, _ = c.client.Topics()
+	}
```

Comment:
this might not solve the issue since if the metadata is false we are retrieving the topics from c.client.Topics() . So the first call will be empty and if the first call is empty if gets all of the metadata and that defeats the purpose .  we have to ignore calling refresh metadata when there no topics present and Metadata.Full is false .